runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -436,10 +436,13 @@ class MM_ObjectAccessBarrierAPI
 	VMINLINE void
 	copyObjectFields(J9VMThread *vmThread, J9Class *objectClass, j9object_t srcObject, UDATA srcOffset, j9object_t destObject, UDATA destOffset)
 	{
+		bool hasReferences = J9CLASS_HAS_REFERENCES(objectClass);
 #if defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER)
-		vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
-#else /* defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER) */
-		if (j9gc_modron_readbar_none != _readBarrierType) {	
+		if (hasReferences) {
+			vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
+		} else
+#endif /* defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER) */
+		if (hasReferences && (j9gc_modron_readbar_none != _readBarrierType)) {
```

Comment:
I'm on a fence even if hasReferences is needed here, but let's leave it.
However, the check was not quite complete. It was missing  '.... || (j9gc_modron_wrtbar_always == _writeBarrierType)' (see how we treat stores)
Please, fix it while here.